spikeinterface/postprocessing/correlograms.py

```diff
@@ -355,42 +315,94 @@ def compute_correlograms_numba(sorting, window_ms: float = 100.0,
     Implementation: Aurélien Wyngaard
     """
 
-    assert HAVE_NUMBA and symmetrize
-    fs = sorting.get_sampling_frequency()
-    num_units = len(sorting.unit_ids)
-
-    window_size = int(round(fs * window_ms/2 * 1e-3))
-    bin_size = int(round(fs * bin_ms * 1e-3))
-    window_size -= window_size % bin_size
+    assert HAVE_NUMBA
+    
     num_bins = 2 * int(window_size / bin_size)
-    assert num_bins >= 1
-
-    bins = np.arange(-window_size, window_size+bin_size, bin_size) * 1e3 / fs
+    num_units = len(sorting.unit_ids)
     spikes = sorting.get_all_spike_trains(outputs='unit_index')
-
     correlograms = np.zeros((num_units, num_units, num_bins), dtype=np.int64)
 
     for seg_index in range(sorting.get_num_segments()):
-        _compute_correlograms_numba(correlograms, spikes[seg_index][0].astype(np.int64),
-                                    spikes[seg_index][1].astype(np.int32),
-                                    window_size, bin_size, fs)
+        spike_times, spike_labels = spikes[seg_index]
+        _compute_correlograms_numba(correlograms,
+                                    spike_times.astype(np.int64),
+                                    spike_labels.astype(np.int32),
+                                    window_size, bin_size)
     
-    return correlograms, bins
+    return correlograms
 
 if HAVE_NUMBA:
-    @numba.jit((numba.int64[:, :, ::1], numba.int64[::1], numba.int32[::1], numba.int32, numba.int32, numba.float32),
+
+    @numba.jit((numba.int64[::1], numba.int32, numba.int32), nopython=True,
+                nogil=True, cache=True)
+    def _compute_autocorr_numba(spike_times, window_size, bin_size):
+        num_half_bins = window_size // bin_size
+        num_bins = 2 * num_half_bins
+        
+        auto_corr = np.zeros(num_bins, dtype=np.int64)
+
+        for i in range(len(spike_times)):
+            for j in range(i+1, len(spike_times)):
+                diff = spike_times[j] - spike_times[i]
+                
+                if diff >= window_size:
+                    break
+
+                bin = int(math.floor(diff / bin_size))
+                #~ auto_corr[num_bins//2 - bin - 1] += 1
+                auto_corr[num_half_bins + bin] += 1
+                #~ print(diff, bin, num_half_bins + bin)
+                
+                bin = int(math.floor(-diff / bin_size))
```

Comment:
I don't think -diff will work for 0.
I remember looking at something like -diff-1 at some point?

Reply:
0 will always be in the same bin. we are not symetric anymore!

Reply:
Ah true!
Forgot this ^^'